File: mobile/src/main/java/com/alexstyl/specialdates/images/AndroidContactsImageDownloader.java
package com.alexstyl.specialdates.images;

import android.content.ContentResolver;
import android.content.Context;
import android.net.Uri;
import android.provider.ContactsContract;

import com.nostra13.universalimageloader.core.download.BaseImageDownloader;

import java.io.FileNotFoundException;
import java.io.InputStream;

public class AndroidContactsImageDownloader extends BaseImageDownloader {

    public AndroidContactsImageDownloader(Context context) {
        super(context);
    }

    @Override
    protected InputStream getStreamFromContent(String imageUri, Object extra) throws FileNotFoundException {
        ContentResolver res = context.getContentResolver();
        Uri uri = Uri.parse(imageUri);

        if (imageUri.startsWith("content://com.android.contacts/")) {
            return ContactsContract.Contacts.openContactPhotoInputStream(res, uri, true);
        }

        return res.openInputStream(uri);

    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/images/AndroidContactsImageDownloader.kt
package com.alexstyl.specialdates.images

import android.content.Context
import android.net.Uri
import android.provider.ContactsContract
import com.nostra13.universalimageloader.core.download.BaseImageDownloader
import java.io.InputStream

class AndroidContactsImageDownloader(context: Context) : BaseImageDownloader(context) {

    override fun getStreamFromContent(imageUri: String?, extra: Any?): InputStream {
        val res = context.contentResolver
        val uri = Uri.parse(imageUri)

        return if (imageUri != null && imageUri.startsWith("content://com.android.contacts/")) {
            ContactsContract.Contacts.openContactPhotoInputStream(res, uri, true)
        } else res.openInputStream(uri)
    }
}
